### PR TITLE
feat(references): RAG + prompt-injection worked example — the remaining B2 coverage slices

### PR DESCRIPTION
**Pull request by bjgreenberg**
## What changed

The two disclosed remaining **B2 coverage slices** (plan finding B2; disclosed in #55): **RAG** and a **prompt-injection worked example**.

- **`references/llm-apps.md`** — new **§7 "RAG — grounding the model in your data"** (design half): escalate into RAG only past in-context limits (rung-1 framing, YAGNI); the retrieval pipeline is deterministic software (pin the embedding model — a silent swap invalidates the index; version chunking; rebuild as a batch job); eval the retriever separately from the generator (labeled query→chunks set, recall@k-style metrics, log chunks used); ground truth or say so (provenance, cited sources, relevance floor with a not-found fallback); the index is a derived cache (re-embed on change, **erasure reaches the vector store**); top-k as cost control. Title/scope updated; §1 forward-pointer. Appended as §7 — **no renumbering**, so the baselined `llm-loop-stopping-criteria` scenario's §3/§5/§6 citations stay valid.
- **`references/secure-data-processing.md`** — two additions:
  - **§2 "Worked example — the two-zone prompt (structural fencing)"**: zone membership decided by who *controls* the text; XML-style fence named in the system prompt; an embedded directive is **reported as a finding, never obeyed**; **planted-delimiter neutralization** (defang look-alikes, whitespace/case-tolerant, visibly); named residual (non-text content blocks sit outside a text fence); fence-integrity tests alongside the injection corpus.
  - **New §4 "RAG & the vector store"** (security half): embeddings/chunks ARE tenant data — an app-side filter after similarity search is not a boundary (structural scoping; RLS-governed vector rows e.g. pgvector, or per-tenant namespaces — with the **ACL-flattening caveat**: a tenant-granularity namespace leaks within the tenant if retrieval ignores document-level ACLs); retrieved chunks are untrusted on every query (**seeded poisoned-corpus proving test** on the retrieval path); **erasure reaches the index**; the embedding call is an egress (DPA/no-train/zero-or-minimal-retention, subprocessor list). Old §4 (OWASP mapping) renumbered to §5 — repo-wide grep confirmed **nothing cited the old §4 by number**.
  - **§5 OWASP mapping updated honestly**: LLM08 mapped to §4 *scoped to the slices actually covered* (embedding inversion + retrieval-activity logging are named residuals); **new LLM09 mapping** to `llm-apps.md` §7's grounding controls (RAG raises misinformation stakes); LLM07 re-scoped from the (wrong) fine-tuning framing to prompt-secrecy hygiene; LLM04 keeps the retrieval-time slice.
- **`SKILL.md`** — both trigger bullets extended (fencing + RAG clauses; the LLM-apps bold label now includes RAG). Byte cost: **92,657 → 93,305 (+648)** — deliberate trigger growth, disclosed.
- **Evals** — 2 new scenarios (suite 40 → **42**): `rag-vector-store-tenant-isolation` (shared-index + app-side-filter + deletion-gap bait) and `prompt-injection-structural-fence` (the "IMPORTANT: ignore instructions" prompt-language-only fix as bait). Both run **PASS** with the skill (results below).
- **Docs same-commit sweep**: README rows + App-layer line + stamp; `threat-modeling-and-api-design.md`'s "three threat surfaces" cross-ref updated to four.

## Why

Plan finding **B2**: LLM-app coverage had no RAG and no concrete prompt-injection worked example — both were explicitly disclosed as the remaining slices when #55 shipped the looping/evals half. House doctrine: each coverage slice lands with SKILL.md trigger + ≥1 eval scenario.

## Testing instructions

- `bash scripts/leakage-guard.sh` → PASS at **Tier 2** (denylist.local present; no Tier-1-only warning).
- Both new scenarios **PASS** with-skill (opus scenario + opus judge, every expected item pass, every anti item clean): `rag-vector-store-tenant-isolation` 1/0/0/0, `prompt-injection-structural-fence` 1/0/0/0 (git-ignored `evals/results/pr1-new-scenario-*`).
- Full with-skill sweep vs the 2026-07-02 baseline runs after the B2 slices land (session plan) — new scenarios baseline on that sweep per `evals/README.md`.
- CI: docs-render, leakage-guard, shellcheck, CodeQL.

## Review

**4-lens adversarial review run pre-PR (source-fidelity, coherence/cross-refs, overclaim, portability/leakage), findings folded in `1a1e015`, fold re-review clean.** One disclosed fix-at-source ride-along from the fold re-review: SKILL.md's data-protection bullet said "zero-retention" where `data-protection.md` (source of truth) says "zero-or-minimal-retention" — aligned (one word, pre-existing divergence). Headline catches folded: "three surfaces" count stale in 3 places (incl. one cross-file); the LLM07/LLM09 "bites when you fine-tune" rationale was factually wrong (LLM09 has a retrieval slice this very PR adds the control for — now a real mapping); "LLM08 implemented" overclaim scoped to slices + named residuals; within-tenant ACL flattening gap; poisoned-corpus rule lacked its proving test; pgvector given the B1 "e.g." marker.

### CODE REVIEW

File: references/secure-data-processing.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
+For any app deployed commercially that **ingests attacker-controlled files, feeds untrusted content to an LLM, and isolates sensitive data across paying tenants**, the surfaces below are non-negotiable. The example multi-tenant SaaS is the worked example — it parses hostile email/office/PDF/archive uploads, sends extracted text to Claude, and stores sensitive records (PII, an append-only audit trail) for many tenants. Four threat surfaces: **the parser, the model, the tenant boundary — and, the moment retrieval is added, the vector store (§4).** Each gets discipline and the tests that prove it (the first three in full QA/test-case structure; §4 folds its tests inline).
```
[the remark this answers is not in the document]
Comment (bjgreenberg) [reply references/secure-data-processing.md]: Fixed in 78bcd8b — restructured to "the tenant boundary — and the vector store, the moment retrieval is added (§4)", dropping the comma-after-"and" construction.
